features/karaf/esh-ext/src/main/feature/feature.xml

```diff
@@ -55,6 +55,17 @@
     <bundle>mvn:org.eclipse.smarthome.binding/org.eclipse.smarthome.binding.yahooweather/${project.version}</bundle>
   </feature>
 
+  <feature name="esh-extensionservice-marketplace" version="${project.version}">
+    <feature>esh-base</feature>
+    <bundle>mvn:org.eclipse.smarthome.extensionservice/org.eclipse.smarthome.extensionservice.marketplace/${project.version}</bundle>
+  </feature>
+
+  <feature name="esh-extensionservice-marketplace-automation" version="${project.version}">
```

Comment:
We do not have feature validation in place at ESH. You are right, it should depend on the marketplace feature.